lib/api-client.js

```diff
@@ -21,47 +27,51 @@ module.exports.getAuthToken = function(config, agentName, socket) {
         , awsami: config.awsami
         , awstype: config.awstype
     };
+
     debug(url);
     debug(postdata);
-    return new Promise(resolve => {
-        request.post(url, { body: postdata, json: true, headers: { 'user-agent': config.UA } }, function(err, res, body) {
-            if (err) {
-                log('Auth error: ' + err);
-                debug(err);
-                if (config.authtimeout) {
-                    clearTimeout(config.authtimeout);
-                }
 
-                config.authtimeout = setTimeout(function() {
-                    resolve(exports.getAuthToken(config, agentName, socket));
-                }, config.AUTHERROR_DELAY);
-                return;
+    return new Promise((resolve) => {
+        return async.retry({
+            times: 5
+            , interval: (retryCount) => {
+                return 100 * Math.pow(2, retryCount);
```

Comment:
maybe make the `100` value a constant so you can adjust it easily?
and why 100ms? do you need it to retry that quickly?